backends/gstreamer/player.rs

```diff
@@ -678,17 +655,17 @@ impl GStreamerPlayer {
         let (receiver, error_handler_id) = {
             let inner_clone = inner.clone();
             let mut inner = inner.lock().unwrap();
-            let pipeline = inner.player.get_pipeline();
+            let pipeline = inner.player.pipeline();
 
             let (sender, receiver) = mpsc::channel();
 
             let sender = Arc::new(Mutex::new(sender));
             let sender_clone = sender.clone();
             let is_ready_clone = self.is_ready.clone();
             let observer = self.observer.clone();
-            let connect_result = pipeline.connect("source-setup", false, move |args| {
+            pipeline.connect("source-setup", false, move |args| {
                 let source = match args[1].get::<gst::Element>() {
-                    Ok(Some(source)) => source,
+                    Ok(source) => source,
```

Comment:
This check is unnecessary. This can simply `unwrap()`. It will only fail if you accidentally connected to a different signal